src/mgmt/srv_pool.c

```diff
@@ -670,21 +731,54 @@ ds_mgmt_pool_target_update_state(uuid_t pool_uuid, uint32_t rank,
 	d_rank_list_t		*ranks;
 	struct mgmt_svc		*svc;
 
+	if (state == PO_COMP_ST_UP) {
```

Comment:
I think this code block should be put in a wrapper function and called it from pool_change_target_state() instead of within this function, because it's very likely you need to reuse it for "server addition", which might not call ds_mgmt_pool_target_update_state()

Reply:
Agreed, when we add the addition component it'll get moved out into its own function. Right now it looks like addition/reintegration will use this function a little differently (because addition will need to do something with the returned target uuids, whereas reintegration only supports reintegrating an existing device with same target uuids)